sycl/doc/extensions/proposed/sycl_ext_oneapi_profiling_tag.asciidoc

```diff
@@ -207,7 +207,7 @@ int main() {
   q.wait();
 
   uint64_t elapsed =
-    end.get_profiling_info<sycl::info::event_profiling::command_start>() -
+    end.get_profiling_info<sycl::info::event_profiling::command_end>() -
```

Comment:
The extension specification says:

> It is unspecified whether the event ever has the `info::event_command_status::running` status, and the meaning of the `info::event_profiling::command_start` timestamp is also unspecified.

Since `end` is the result of a profiling tag submission, I assumed this was a mistake in the example as the unspecified meaning of `command_start` partially also leaves the difference unspecified as well.

Given the extension also encourages `command_start` to be the same as `command_end`, it seems like the change is both better defined and (for implementations following the suggestions) resulting in the same elapsed time.

Reply:
This was not our intention when we wrote this spec, but I agree that the wording here should be changed.  Our intention was to allow either of two possible implementation choices:

* An implementation where the profiling tag does not take any time to execute.  In this case the `command_start` and `command_end` timestamps would be the same.  This is the preferred implementation if it is possible.
* An implementation that submits an empty kernel to the queue.  In this case the `command_end` timestamp would be slightly greater than the `command_start` timestamp.

We intentionally wanted to show an example where the elapsed time is computed as "end::command_start - start::command_end" because that will exclude the time of the empty kernel if an implementation chooses the second approach.

The relevant discussion is [this comment](https://github.com/intel/llvm/pull/12194#discussion_r1428509659) in #12194.

I propose that we change the wording of the spec to say this instead:

> It is unspecified whether the event ever has the `info::event_command_status::running` status, and the meaning of the `info::event_profiling::command_start` timestamp is not precisely specified. Implementations are encouraged to transition the event directly from the "submitted" status to the "complete" status and are encouraged to set the "command_start" timestamp to the same value as the "command_end" timestamp.  However, some implementations may set the "command_start" timestamp to a value that is slightly less than "command_end" if the implementation of the profiling tag takes non-zero time to execute.

Tagging @Pennycook here also since he was involved with the original discussion.

Reply:
Seems reasonable to me, but I still think the wording is a little to imprecise. I have reverted the change to the spec and added tests for start that should at least adhere to the intention, but I think it would be better to make the changes to the extension in a separate patch so we can discuss the wording.